src/Text/blocks/text.jl

```diff
@@ -0,0 +1 @@
+struct TextBlock <: Block end
```

Comment:
Some more documentation on what this represents would be great. Is this a sentence, a paragraph? A `checkblock` is also needed. The convention is not to put `Block` in the name of `Block`s, so depending on what this represents, we may be able to find a more precise term.

What tasks does this adress? Is there a need for different text-related blocks to solve other tasks? I'm. not super familiar with deep NLP, but these questions may help making this more precise.

Reply:
> Some more documentation on what this represents would be great. Is this a sentence, a paragraph?

The initial idea was to replicate `struct Image{N} <: Block end`. I haven't specified the dimension `N` as I'm not sure how to incorporate it with textual data is it the length? Can that be the vocabulary size? Or can we just leave this as it is? (Maybe we can discuss this). 

> Is there a need for different text-related blocks to solve other tasks?

Most text-related tasks revolve around classification and sentiment analysis (Which are single label prediction tasks). There are some more tasks that involve predicting multiple labels like topic modeling (Need more than one label to be predicted). Finally, there are some unsupervised techniques like text generation, text summarization.

I guess all the observations for the above tasks can be represented using a block of the same type?

> Is this a sentence, a paragraph?

I think the better name would be `Paragraph` rather than `TextBlock` as most if not all of the text data has a single paragraph (multiple sentences) as an individual observation (We can consider a sentence as a paragraph too!).

Reply:
>  A checkblock is also needed.

I tried adding the `checkblock` for `TextBlock`, but it is always returning false. I guess it is calling the default `checkblock` method. I'm sure why this happens. Also the method I wrote isn't even getting displayed in the output of `methods(checkblock)`.

Reply:
> The initial idea was to replicate struct Image{N} <: Block end. I haven't specified the dimension N as I'm not sure on incorporating it with textual data. Is it the length? Can that be the vocabulary size? Or can we just leave this as it is? (Maybe we can discuss this).

Don't think this is necessary. As described in the documentation of `Block`, the block should only hold information that is constant for all observations, so encoding vocab size here would mean we assume it is fixed for all observations in a dataset which is too restrictive.

> I guess all the observations for the above tasks can be represented using a block of the same type?

Yeah, that sounds like it 👍 

> I think the better name would be Paragraph rather than TextBlock as most if not all of the text data has a single paragraph (multiple sentences) as an individual observation (We can consider a sentence as a paragraph too!).

`Paragraph` is good :)

Reply:
> Paragraph is good :)

Yes, that's what I thought would be more reasonable. Changed it now.